File: pcpartsbackend/src/main/java/erp/pcpartsbackend/models/Order.java
package erp.pcpartsbackend.models;

import com.fasterxml.jackson.annotation.JsonFormat;
import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import org.hibernate.annotations.OnDelete;
import org.hibernate.annotations.OnDeleteAction;

import java.io.Serializable;
import java.util.Date;
import java.util.UUID;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Table(name="orders")
public class Order implements Serializable {
    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private UUID orderId;
    private Date orderDate;
    private Float orderPrice;
    private String orderStatus;
    private double discount;
    private String promoCode;
    @ManyToOne(fetch = FetchType.LAZY)
    @OnDelete(action = OnDeleteAction.SET_NULL)
    @JoinColumn(name = "customer_id")
    private Customer customer;
}

File: pcpartsfrontend/pcpartsshop/src/app/models/Order.ts
import {ProductOrder} from "./ProductOrder";
import {Customer} from "./Customer";

export class Order {
  constructor(
    public orderId: string,
    public orderDate: Date,
    public orderPrice: number,
    public orderStatus: string,
    public discount: number,
    public promoCode: string,
    public customer: Customer,
    public productOrders: ProductOrder[]
  ) {}
}

export interface OrderCreateRequest {
  orderDate: Date;
  orderPrice: number;
  orderStatus: string;
  discount: number;
  promoCode?: string;
  customerId: string;
}
